lib/apia/endpoint.rb

```diff
@@ -48,10 +48,24 @@ def execute(request)
         environment = RequestEnvironment.new(request, response)
 
         catch_errors(response) do
-          # Determine an authenticator and execute it before the request happens
+          # Determine an authenticator for this endpoint
           request.authenticator = definition.authenticator || request.controller&.definition&.authenticator || request.api&.definition&.authenticator
+
+          # Execute the authentication before the request happens
           request.authenticator&.execute(environment)
 
+          # Add the CORS headers to the response before the endpoint is called. The endpoint
+          # cannot influence the CORS headers.
+          response.headers.merge!(environment.cors.to_headers)
+
+          # OPTIONS requests can always reeturn 200 OK and no body. They will also contain the CORS
+          # headers that have been determined from the endpoint or the authenticator.
```

Comment:
Nitpicks: The comment right above this one says the endpoint cannot determine CORS headers, but this comment seems to suggest that endpoints can.

Reply:
That comment is incorrect 😄 I shall fix.